src/legacy/core_plugins/kbn_vislib_vis_types/public/utils/collections.ts

```diff
@@ -81,10 +84,11 @@ const getChartTypes = () => [
   },
 ];
 
-export enum ChartModes {
```

Comment:
What's wrong with an enum here?

Reply:
The short answer, nothing really.

In elastic-charts we only use object enums because of the bloat the standard ts `enums` create. https://github.com/elastic/elastic-charts/pull/266

The regular enum would produce something like

```ts
enum Pet1 {
  Cat = 'Cat',
  Dog = 'Dog',
}

// transpiles to this
var Pet1;
(function (Pet1) {
    Pet1["Cat"] = "Cat";
    Pet1["Dog"] = "Dog";
})(Pet1 || (Pet1 = {}));
```

where a much cleaner output is 

```ts
const Pet2 = Object.freeze({
  Cat: 'Cat' as 'Cat',
  Dog: 'Dog' as 'Dog',
})
type Pet2 = typeof Pet2.Cat | typeof Pet2.Dog;

// transpiles to this
const Pet2 = Object.freeze({
    Cat: 'Cat',
    Dog: 'Dog',
});
```

Using a `const enum` would be clearer but they are not yet supported in babel (https://github.com/babel/babel/issues/8741)